readthedocs/projects/tasks.py

```diff
@@ -1416,6 +1409,27 @@ def warn(self, msg):
     delete_queryset.delete()
 
 
+@app.task(max_retries=3, default_retry_delay=7 * 60)
+def clean_build_task(version_id):
+    """Clean the files used in the build of the given version."""
+    version = Version.objects.get_object_or_log(pk=version_id)
+    if (
+        not version or
+        not version.project.has_feature(Feature.CLEAN_AFTER_BUILD)
+    ):
+        return
```

Comment:
I'd return `False` here indicating that the cleanup was not done and `True` after `remove_dirs`.

This helps debugging and understanding what's happening.

Reply:
Also, this if block could log something saying that the clean was skipped